### ISSUE TITLE
Infer shape error

### ISSUE THREAD

**Issue by patwa67** [dmlc/mxnet#6630]
My model below works fine when I set num_hidden=1, but not when I set num_hidden=2. 
I get the error:
infer_shape error. Arguments:
data: (150, 9723L, 3L)
lin_reg_label: (150,)

Any suggestions how to get around the error?

```
import mxnet as mx
import numpy as np

from numpy import genfromtxt
my_data = genfromtxt('Data.csv', delimiter=',')

label = my_data[:,0]-np.mean(my_data[:,0])
ntrain = 2326
traindata = my_data[:ntrain, 1:9724]
testdata = my_data[ntrain:, 1:9724]

hottrain = mx.nd.one_hot(mx.nd.array(traindata, dtype=np.int32), 3)
hottest = mx.nd.one_hot(mx.nd.array(testdata, dtype=np.int32), 3)

del my_data

batch_size = 150

train_iter = mx.io.NDArrayIter(hottrain, label[:ntrain], batch_size, shuffle=True,label_name='lin_reg_label')
eval_iter = mx.io.NDArrayIter(hottest, label[ntrain:], batch_size)

X = mx.sym.Variable('data')
Y = mx.symbol.Variable('lin_reg_label')
dropout = mx.symbol.Dropout(data=X, name='do1', p = 0.5)
fully_connected_layer  = mx.sym.FullyConnected(data=dropout, name='fc1', num_hidden = 2)
lro = mx.sym.LinearRegressionOutput(data=fully_connected_layer, label=Y, name="lro")

model = mx.mod.Module(
    symbol = lro ,
    data_names=['data'],
    label_names = ['lin_reg_label']
)

model_prefix = 'mx_reg'
checkpoint = mx.callback.do_checkpoint(model_prefix)

model.fit(train_iter, eval_iter, optimizer='Adam',
            optimizer_params={'wd':1.3,'learning_rate':0.0001},
            num_epoch=100,
            epoch_end_callback=checkpoint)
```

**Comment by piiswrong** [comment]
@reminisce

**Comment by patwa67** [comment]
And there will still be only one loss function?

**Comment by patwa67** [comment]
I changed to:
```
yc = np.array(my_data[:,0]-np.mean(my_data[:,0]))
label = np.transpose(np.tile(yc,(2,1)))
```

and then ran with num_hidden = 2, but this just produces two copies of the num_hidden = 1 case.

**Comment by chunyang-wen** [comment]
[Symbol-FullyConnected](http://mxnet.io/api/python/symbol.html#mxnet.symbol.FullyConnected)

num_hidden (int, required) – Number of hidden nodes of the output. So when set num_hidden = 2, you are expecting two outputs instead of one for each data instance.

**Comment by patwa67** [comment]
I thought num_hidden = 2 meant two nodes of one layer, and that these nodes could be connected to a one unit output without adding one more layer. This is obviously not the case. It can be solved by adding an extra layer with num_hidden = 1.
